### PR TITLE
options to distinguish internet, local network and localhost

### PR DESCRIPTION
**Pull request by hgourvest**
- removed « fenced » option, use « localhost » instead (bidirectionnal)
- removed general LAN option, you can now activate this option
individually (bidirectionnal).
- bypass option no longer give access to LAN and localhost, you must
check localhost and local network option to have the old behavior.
- various cleanup, attempt to fix UI bugs …

### CODE REVIEW

**Comment by hgourvest** [comment]
BTW syncthings app is now working fine with localhost and localnetwork options

**Comment by hgourvest** [comment]
also, it will fix this bug: https://github.com/EthACKdotOrg/orWall/issues/83
sometimes, subnet change was not applyed to iptables

File: app/src/main/java/org/ethack/orwall/database/natDBHelper.java [hunk 1]
```diff
@@ -16,22 +16,23 @@
...
                     NAT_TABLE_NAME,
                     COLUMN_APPUID,
                     COLUMN_APPNAME,
-                    COLUMN_ONIONTYPE, Constants.DB_ONION_TYPE_TOR,
-                    COLUMN_ONIONPORT, Constants.ORBOT_TRANSPROXY,
-                    COLUMN_PORTTYPE
+                    COLUMN_ONIONTYPE,
+                    COLUMN_LOCALHOST,
+                    COLUMN_LOCALNETWORK
             );
```
Comment (cjeanneret) [inline app/src/main/java/org/ethack/orwall/database/natDBHelper.java]: Hello,

I'm concerned about backward compatibility: will current rules still work properly and be migrated to this new DB schema? Else, we should find a way to alert people "hey, you rules will be dropped with this update" I guess.
I think that's possible by updating line 37 value of DATABASE_VERSION is this very file.

I remember there is some DB migration capabilities… Care to check?

**Comment by cjeanneret** [comment]
Heya!

That's a huge commit :+1:

The only concern I have so far is the database content — might be good to dig a bit in the docs in order to migrate it. If I'm not wrong, with the current code, only an app removal followed by a fresh install will "migrate" the DB schema — of course, that's not a good idea for "old timers".

For the rest, I don't see any bad things.

Cheers,

C.

**Pull request closed without merging**